src/Tracer.php

```diff
@@ -0,0 +1,203 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Exercism\PhpTestRunner;
+
+use Exercism\PhpTestRunner\Result;
+use PHPUnit\Event\Code\TestMethod;
+use PHPUnit\Event\Event;
+use PHPUnit\Event\Test\BeforeFirstTestMethodErrored;
+use PHPUnit\Event\Test\Errored;
+use PHPUnit\Event\Test\Failed;
+use PHPUnit\Event\Test\Passed;
+use PHPUnit\Event\Test\PrintedUnexpectedOutput;
+use PHPUnit\Event\TestRunner\Finished;
+use PHPUnit\Event\Tracer\Tracer as TracerInterface;
+use ReflectionClass;
+
+final class Tracer implements TracerInterface
```

Comment:
Should we favor tracer instead of subscribers? (https://docs.phpunit.de/en/10.5/extending-phpunit.html#implementing-an-event-subscriber)

I think subscribers would be more performant.

Reply:
Each subscriber is its own class. So one needs many classes forwarding the information received to a data collector class. And there is no way to simply "sniff" all events to spot the right one for a new situation.

Performance did not change significantly when I experimented with subscribers. I only had a more complex class hierarchy and problems to trace the events order.